Self-Evolving-Car/Self-Evolving-Bot/Self-Evolving-Bot.py

```diff
@@ -0,0 +1,240 @@
+import math
+import random
+import numpy
+from rlbot.utils.game_state_util import GameState, BallState, CarState, Physics, Vector3, Rotator, GameInfoState
+from rlbot.agents.base_agent import BaseAgent, SimpleControllerState
+from rlbot.utils.structures.game_data_struct import GameTickPacket
+
+
+class PythonExample(BaseAgent):
+
+    def initialize_agent(self):
+        self.controller_state = SimpleControllerState()
+        self.frame = 0  # frame counter for timed reset
+        self.brain = -1  # bot counter for generation reset
+        self.pop = 10  # population for bot looping
+        self.out = [None] * self.pop  # output of nets
+        self.gen = 0
+        self.botList = [] #list of Individual() objects
+        self.fittest = Fittest() #fittest object
+        self.pos = 2500 #ball position
+        self.mutRate = 0.1 #mutation rate
+        self.distance_to_ball = [10000]*10000 #set high for easy minumum
+        self.nodeNum = 7 #number of nodes for Node() object array for each individual
+
+        #CREATE BOTS AND NETS
+        for i in range(self.pop):
+            self.botList.append(Individual())
+            self.botList[i].create_node()
+            self.botList[i].name = "Bot "+str(i)
+
+        #ASSIGN WEIGHTS
+        for i in self.botList:
+            print("INIT: "+i.name)
+            for p in range(0,len(i.weights)):
+                i.weights[p] = random.uniform(-self.mutRate,self.mutRate)
+                print("----"+str(i.weights[p]))
+
+
+    def get_output(self, packet: GameTickPacket) -> SimpleControllerState:
+
+        #INIT LOOPS
+        if self.frame == 0:
+            self.calcFitness()
+            self.reset() #reset at start
+            self.brain += 1 #change bot every reset
+        if self.brain >= self.pop:
+            self.gen += 1
+
+            #PRINT GENERATION INFO
+            self.avgBestFitness()
+            print("")
+            print("     GEN = "+str(self.gen))
+            print("-------------------------")
+            print("FITTEST = "+str(self.botList[self.calcFittest()].name))
+            print("------FITNESS = " + str(self.fittest.fitness))
+            print("------WEIGHTS = " + str(self.botList[self.fittest.index].weights))
+            for i in range(len(self.botList)):
+                print("FITNESS OF BOT "+str(i)+" = "+str(self.botList[i].fitness))
+
+            #NE Functions
+            self.selection()
+            self.mutate()
+            self.brain = 0 #reset bots after all have gone
+
+        self.frame = self.frame + 1
+
+
+        #NEURAL NET INPUTS
+        ball_location = Vector2(packet.game_ball.physics.location.x, packet.game_ball.physics.location.y)

```

Comment:
Put this into a separate function. We have a class designed for inputs so looking into that would be good